### PR TITLE
Set globally Java properties for ScalaCLI launcher

### PR DESCRIPTION
**Pull request by lwronski**
Enable users to globally set Java properties for ScalaCLI launcher using the `config` command.

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/commands/config/Config.scala [hunk 1]
```diff
@@ -298,4 +300,32 @@ object Config extends ScalaCommand[ConfigOptions] {
       }
     }
   }
+
+  /** If the new value is the same as the previous value, ask user for confirmation or suggest to
```
Comment (MaciejG604) [inline modules/cli/src/main/scala/scala/cli/commands/config/Config.scala]: Something's mixed up in this scaladoc - if values are the same then no confirmation nor --force flag is needed, right?

Comment (lwronski) [reply modules/cli/src/main/scala/scala/cli/commands/config/Config.scala]: Right! Good catch.

File: modules/cli/src/main/scala/scala/cli/commands/config/Config.scala [hunk 2]
```diff
@@ -285,6 +278,15 @@ object Config extends ScalaCommand[ConfigOptions] {
                       else
                         values
 
+                    val previousValue = db.get(entry).wrapConfigException.orExit(logger)
+
+                    confirmUpdatedValue(
```
Comment (MaciejG604) [inline modules/cli/src/main/scala/scala/cli/commands/config/Config.scala]: I think this makes almost all config writes interactive now (except for `Keys.repositoryCredentials` and `Keys.publishCredentials`). We could limit that to only Java props key, or only to `StringListEntry` keys.

IMO for keys that have just a single value, so our `BooleanEntry`, `PasswordEntry` or `StringEntry`, asking for confirmation is not that useful.
WDYT?

File: website/docs/under-the-hood.md [hunk 3]
```diff
@@ -46,6 +46,11 @@ In order set them, the `-D` command-line flags must be placed as the first optio
 scala-cli -Dfoo1=bar1 -Dfoo2=bar2 run ...
 ```
 
+:::note
+- `scala-cli run . -Dfoo=bar` would pass the java property into your Scala app
+- `scala-cli -Dfoo=bar run .` would pass the java property into `scala-cli.`
```
Comment (MaciejG604) [inline website/docs/under-the-hood.md]:
```suggestion
- `scala-cli -Dfoo=bar run .` would pass the java property into `scala-cli`
```

File: website/docs/under-the-hood.md [hunk 4]
```diff
@@ -59,7 +64,15 @@ $ cat .scala-jvmopts
 $ scala-cli run ...
 ```
 
+In the Scala CLI, you can set Java properties globally for the Scala CLI launcher using the `config` command.
```
Comment (MaciejG604) [inline website/docs/under-the-hood.md]:
```suggestion
You can set Java properties globally for the Scala CLI launcher using the `config` command.
```

File: website/docs/under-the-hood.md [hunk 5]
```diff
@@ -59,7 +64,15 @@ $ cat .scala-jvmopts
...
+The example belows shows how to set the Java properties `-Djavax.net.ssl.trustStore=cacerts` and `-Dfoo=bar2`:
```
Comment (MaciejG604) [inline website/docs/under-the-hood.md]:
```suggestion
The example below shows how to set the Java properties `-Djavax.net.ssl.trustStore=cacerts` and `-Dfoo=bar2`:
```

File: website/docs/under-the-hood.md [hunk 6]
```diff
@@ -59,7 +64,15 @@ $ cat .scala-jvmopts
...
+```
+
 :::note
-- `scala-cli run . -Dfoo=bar` would pass the java property into your Scala app
-- `scala-cli -Dfoo=bar run .` would pass the java property into `scala-cli.`
-:::
\ No newline at end of file
+Please note that if you need to modify the Java properties, you have to redefine all the properties. It's not possible
+to update just a single Java properties via the `config` command. Each update effectively replaces the entire Java properties
+list.
```
Comment (MaciejG604) [inline website/docs/under-the-hood.md]:
```suggestion
Please note that if you need to modify the Java properties, you have to redefine all of them. It's not possible
to update just a single value via the `config` command. Each update effectively replaces the entire Java properties list.
```

File: modules/cli/src/main/scala/scala/cli/commands/config/Config.scala (hunk 2 above)
Comment (lwronski) [reply modules/cli/src/main/scala/scala/cli/commands/config/Config.scala]: Yes, that sounds like a good idea. so, I limit it only to `StringListEntry`.

File: website/docs/under-the-hood.md (hunk 3 above)
Comment (MaciejG604) [reply website/docs/under-the-hood.md]: I think this is still unresolved

**Review by MaciejG604 — APPROVED** [review]

**Pull request merged**